src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,83 @@ Streaming.prototype.disposeSubscription = function(subscription) {
     }
 };
 
+function createHandleSubscriptionReadyForUnsubscribe(subscriptions, resolve) {

```

Comment:
how about a top level function

handleSubscriptionReadyForUnsubscribe

taking the 2 arguments,

then to "create" the one to pass in, use bind